### PR TITLE
Allow kubernete context to be supplied #554

### LINKED ISSUE #554: kubectl backend should support contexts
It appears to only support the following options:
- `container`
- `namespace`
- `kubeconfig`

Eg: 
```
py.test --hosts='kubectl://somepod-2536ab?container=nginx&namespace=web&kubeconfig=/path/kubeconfig'
```

My use case heavily relies on selecting contexts (but not switching the current context).

This would be ideal:

```
py.test --hosts='kubectl://somepod-2536ab?container=nginx&context=dev-sydney'
```

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
I have not run the tests as I couldn't work out how to install the deps from my locked down environment.

Fixes #554

### CODE REVIEW

File: test/test_backends.py [hunk 1]
```diff
@@ -445,7 +445,8 @@ def test_parse_hostspec(hostspec, expected):
     ('kubectl://pod', 'pod', None, None, None),
     ('kubectl://pod?namespace=n', 'pod', None, 'n', None),
     ('kubectl://pod?container=c&namespace=n', 'pod', 'c', 'n', None),
-    ('kubectl://pod?namespace=n&kubeconfig=k', 'pod', None, 'n', 'k')
+    ('kubectl://pod?namespace=n&kubeconfig=k', 'pod', None, 'n', 'k'),
+    ('kubectl://pod?context=ctx&container=c', 'pod', 'ctx', 'c', None)
```
Comment (philpep) [inline test/test_backends.py:449]: you have to add a "context" parameter in `pytest.mark.parametrize` and in the function arguments and then `assert backend.context == context`

Comment (NickLarsenNZ) [reply test/test_backends.py]: Ohh, just put in some updates once I understood how. I hadn't done the assertion bit. I'll take a look now.

Thanks for your speedy reply @philpep

Comment (NickLarsenNZ) [reply test/test_backends.py]: Done this time (I'll stick around to make sure the test passes)

Comment (NickLarsenNZ) [reply test/test_backends.py]: Just trying to tackle the linter :/

Comment (NickLarsenNZ) [reply test/test_backends.py]: @philpep I think the 80 char limit is a bit too narrow. When combined with the indentation rule, it's never ending trying to satisfy both constraints.

Are you able to suggest how I should layout the test? Or can we look at relaxing the rules (might lower developer burden - or maybe it's just me).

File: test/test_backends.py [hunk 2]
```diff
@@ -443,19 +443,21 @@ def test_parse_hostspec(hostspec, expected):
...
-    assert backend.context == context
+        ('kubectl://pod', 'pod', None, None, None, None),
+        ('kubectl://pod?namespace=n', 'pod', None, 'n', None, None),
+        ('kubectl://pod?container=c&namespace=n', 'pod', 'c', 'n', None, None),
+        ('kubectl://pod?namespace=n&kubeconfig=k', 'pod', None, 'n', 'k', None),
+        ('kubectl://pod?context=ctx&container=c', 'pod', 'c', None, None, 'ctx')
+    ]
+)
+def test_kubectl_hostspec(hostspec, pod, container, namespace, kubeconfig, 
+    context):
```
Comment (philpep) [inline test/test_backends.py:454]: Yes python format rules and related tools are a bit boring... Try this:

```suggestion
def test_kubectl_hostspec(
    hostspec, pod, container, namespace, kubeconfig, context,
):
```

File: test/test_backends.py [hunk 3]
```diff
@@ -441,18 +441,24 @@ def test_parse_hostspec(hostspec, expected):
...
-    assert backend.container == container
-    assert backend.namespace == namespace
-    assert backend.kubeconfig == kubeconfig
+@pytest.mark.parametrize(
+    'hostspec,pod,container,namespace,kubeconfig,context', [
+        ('kubectl://pod', 'pod', None, None, None, None),
+        ('kubectl://pod?namespace=n', 'pod', None, 'n', None, None),
+        ('kubectl://pod?container=c&namespace=n', 'pod', 'c', 'n', None, None),
+        ('kubectl://pod?namespace=n&kubeconfig=k', 'pod', None, 'n', 'k', None),
+        ('kubectl://pod?context=ctx&container=c', 'pod', 'c', None, None, 'ctx')
```
Comment (philpep) [inline test/test_backends.py:450]:
```suggestion
        ('kubectl://pod?container=c&namespace=n',
         'pod', 'c', 'n', None, None),
        ('kubectl://pod?namespace=n&kubeconfig=k',
         'pod', None, 'n', 'k', None),
        ('kubectl://pod?context=ctx&container=c',
         'pod', 'c', None, None, 'ctx')
```

File: test/test_backends.py [hunk 4]
```diff
@@ -441,18 +441,24 @@ def test_parse_hostspec(hostspec, expected):
...
+)
+def test_kubectl_hostspec(
+    hostspec, pod, container, namespace, kubeconfig, context,
+):
+        backend = testinfra.get_host(hostspec).backend
+        assert backend.name == pod
+        assert backend.container == container
+        assert backend.namespace == namespace
+        assert backend.kubeconfig == kubeconfig
+        assert backend.context == context
```
Comment (philpep) [inline test/test_backends.py:461]:
```suggestion
    backend = testinfra.get_host(hostspec).backend
    assert backend.name == pod
    assert backend.container == container
    assert backend.namespace == namespace
    assert backend.kubeconfig == kubeconfig
    assert backend.context == context
```

**Comment by philpep** [comment]
Merged, thanks!

**Comment by NickLarsenNZ** [comment]
@philpep will this be available via Pypi soon? Only 5.1.0 (Released: May 18, 2020) is showing up on pypi.org

**Comment by philpep** [comment]
@NickLarsenNZ sure, I just uploaded 5.2.1 including this. Happy upgrade!

**Pull request merged**